distribution/lib/Standard/Visualization/0.0.0-dev/src/Scatter_Plot.enso

```diff
@@ -133,23 +134,78 @@ Table.Table.axes self =
 
 ## PRIVATE
 Vector.Vector.point_data : Vector -> Object
-Vector.Vector.point_data self =
-        self.map_with_index <| i-> elem->
-            Json.from_pairs [[X.name,i],[Y.name,elem]]
+Vector.Vector.point_data self  =
+    self.map_with_index <| i-> elem->
+        Json.from_pairs [[X.name,i],[Y.name,elem]]
 
 ## PRIVATE
-json_from_table : Table -> Json
-json_from_table table =
-    data = table.point_data
+bound_data bounds data = case bounds of
+    Nothing -> data
+    _ ->
+        min_x = bounds.at 0
+        min_y = bounds.at 1
+        max_x = bounds.at 2
+        max_y = bounds.at 3
+
+        data.filter <| datum->
+            x = datum.get "x" . unwrap
+            y = datum.get "y" . unwrap
+
+            min_x<=x && x<=max_x && min_y<=y && y<=max_y
+
+type Extreme min_x max_x min_y max_y
+
+## PRIVATE
+limit_data limit data = case limit of
+    Nothing -> data
+    _ -> if (data.length <= limit) then data else
+        b = Vector.Builder.new
```

Comment:
Use
`Vector.new_builder limit`
to create the builder.

Reply:
Done in [42cbf76](https://github.com/enso-org/enso/pull/3655/commits/42cbf76e27eff2d19d2869098c7afc7c86cb052c)